cylc/flow/cycling/iso8601.py

```diff
@@ -432,6 +405,7 @@ def set_offset(self, i_offset):
         if self.exclusions:
             self.value += '!' + str(self.exclusions)
 
+    @lru_cache(100)
```

Comment:
This is the only functional change, caching up to 100 values for the `is_on_sequence` method.